rollbar/__init__.py

```diff
@@ -788,6 +837,9 @@ def _report_message(message, level, request, extra_data, payload_data):
     _add_lambda_context_data(data)
     data['server'] = _build_server_data()
 
+    if _tags.value:
+        data['tags'] = _flatten_nested_lists(_tags.value)
```

Comment:
no, `_tags.value` at this point would look something like
```
[
  [
    {'key': 'feature_flag.name', 'value': 'feature-1'},
    {'key': 'feature_flag.data.feature-1.order', 'value': 0}
  ],
  [
    {'key': 'feature_flag.name', 'value': 'feature-2'},
    {'key': 'feature_flag.data.feature-2.order', 'value': 1},
    {'key': 'feature_flag.data.feature-2.variation', 'value': True}
  ]
]
```
where each inner list keeps the tags used to represent a feature flag together. I thought this was necessary because when an exception occurs in the scope of a context manager and we pop and attach the tags to the exception.

Reply:
Ok I see. I think we need to change the tag manager so it appends each tag to the list:

```
class _TagManager(object):
    """
    Context manager object that interfaces with the `_tags` stack:
        On enter, puts the tag object at top of the stack.
        On exit, pops off the top element of the stack.
          - If there is an exception, attach the tag object to the exception
            for rebuilding of the `_tags` stack before reporting.
    """
    def __init__(self, tag):
        self.tag = tag

    def __enter__(self):
        _tags.append(self.tag)
```

That `self.tag` should be `self.tags` and we shouldn't do `append()` but maybe `extend()` so `_tags` is always flatten

Reply:
I'm not sure I agree with using `extend` or keeping `_tags` always flatten. In the `__exit__` method:
```
    def __exit__(self, exc_type, exc_value, traceback):

        if exc_value:
            if not hasattr(exc_value, '_rollbar_tags'):
                exc_value._rollbar_tags = []

            exc_value._rollbar_tags.append(self.tag)

        _tags.pop()
```
we pop from `_tags` whenever we exit the context manager. If this was flattened, we'd have to keep track of the number of times we should pop.

Well, I guess this is something we can do. Would something like this be preferred?
```
    def __exit__(self, exc_type, exc_value, traceback):

        if exc_value:
            if not hasattr(exc_value, '_rollbar_tags'):
                exc_value._rollbar_tags = []

            exc_value._rollbar_tags.append(self.tag)

        for tag in self.tag:
            _tags.pop()
```

Reply:
Oh ok, I see what you mean. You are right although maybe we can:

- rename `_tags` to something different that represents better that they are "context tags" or similar
- Add a method in `_LocalTags` that gives you the flatten tags. That way `_tags` will hide the internal structure and I think the code that uses it can be cleaner since will not deal with flattening the list, etc...

What you think?

Reply:
Oooo I like that! Let me see if I cam implement it :D

Reply:
done